### DISCUSSION TITLE
using `Unicorn.call` in multiple child elements

### DISCUSSION THREAD

**Discussion by jacksund** [django-commons/django-unicorn#657] [Q&A] [answered]
I have a form that uses a select2 dropdown menu, where this menu is within a child element.

So my setup is essentially the doc's example with [multiple-children](https://www.django-unicorn.com/docs/child-components/#multiple-children) plus the [ignore-elements](https://www.django-unicorn.com/docs/templates/#ignore-elements) when editing one row (specifically one column in table will have a select2 element).

How do I pass the key to `Unicorn.call`? Would it be something like this:
``` js
Unicorn.call('row', 'example_key', 'select_state', this.value, this.selectedIndex)
```

Or is passing the key not necessary?

**Comment by jacksund** [comment, marked as answer]
I think I found the solution. The first arg in `Unicorn.call` is `componentNameOrKey`:

https://github.com/adamghill/django-unicorn/blob/dca46768e031057cb764f36ae23652d570968d6b/django_unicorn/static/unicorn/js/unicorn.js#L130

So you can call the child component as follows:
``` javascript
Unicorn.call('example_key', 'select_state', this.value, this.selectedIndex)
```

**Reply by adamghill** [reply]
Yep, that should work!
